code/modules/religion/rites.dm

```diff
@@ -0,0 +1,55 @@
+/datum/religion_rites
+	var/name = "religious rite"
+	var/desc = "immm gonna rooon"
+	var/ritual_length = (10 SECONDS) //total length it'll take
+	var/list/ritual_invocations //strings that are by default said evenly throughout the rite
+	var/favor_cost = 0
+
+///Called to perform the invocation of the rite, with args being the performer and the altar where it's being performed. Maybe you want it to check for something else?
+/datum/religion_rites/proc/perform_rite(mob/living/user, obj/structure/altar_of_gods/AOG)
+	if(GLOB.religious_sect?.favor < favor_cost)
+		to_chat(user, "<span class='warning'>This rite requires more favor!</span>")
+		return FALSE
+	to_chat(user, "<span class='notice'>You begin to perform the rite of [name]...</span>")
+	if(!ritual_invocations)
+		if(do_after(user, target = user, delay = ritual_length))
+			return TRUE
+		return FALSE
+	var/c_c_combo = 0
+	for(var/i in ritual_invocations)
+		addtimer(CALLBACK(user, /atom/movable/proc/say, i), (ritual_length/ritual_invocations.len)*c_c_combo) //first one is instant
```

Comment:
what if you stop doing the ritual halfway through?

Reply:
should there be a check to see if you can speak, or do you think that that would just be annoying/restrict potential chaplain gimmicks?

Reply:
also, the do_after of the ritual itself might take less time than you think it will, since having a good sanity level decreases the amount of time that it takes to perform do_after()s, IIRC